### DISCUSSION TITLE
Specify redirect url when authenticating users with magic link.

### DISCUSSION THREAD

**Discussion by spencerbart** [supabase/supabase#6943] [Feature Requests]
Currently, only OAuth providers can sign in with redirect where it can take an optional param called `redirectTo` that specifies where to redirect them to. It would be awesome if regular Magic Link auth could take this param and use it when generating a link.
![image](https://user-images.githubusercontent.com/38776747/169879111-7274eb65-581c-448b-a1d9-4d1d52e340c7.png)

We regularly have people try to access a resource (a custom portal where the url is sent to them in this case) that is authenticated by their email. We want the magic link to redirect back to that resource and not to a default url because the resource changes from user to user.

Like this
![image](https://user-images.githubusercontent.com/38776747/169879464-f68f49cc-5c54-47ac-aeb2-12215226782b.png)

I have looked through the gotrue repo and it seems like it's only a few lines of code that need to be changed to make this work.

**Comment by spencerbart** [comment]
This seems like super low-hanging fruit. Most of the functionality is already there. A new param added to `MagicLinkParams`, check to see if it's there, and then pass the prop down to the`sendMagicLink` function.

This is a screenshot from `magic_link.go` file in the gotrue repo.
![Screen Shot 2022-05-23 at 2 01 33 PM](https://user-images.githubusercontent.com/38776747/169896884-15f7de7d-12ec-4605-b1ab-ac8a9bfbaacc.png)

**Reply by spencerbart** [reply]
This could be a super easy build and I would love to build it myself, I just haven't had any experience with writing code for an open source repo. If any contributor to Supabase wants to take this on, I could write the code, I just need someone to oversee the code review and championing it to production. I am unfamiliar with the open source pipeline.

**Reply by dallinbentley** [reply]
This would be super helpful for what we are trying to do.  Looked at the code and seems like it might be pretty straightforward.  Willing to contribute to make this happen.

**Comment by spencerbart** [comment]
Update: this works but you have to specify redirect URLs in the Supabase Dashboard. There is also no documentation for this. All that needs to be done is the documentation.

<img width="1704" alt="Screen Shot 2022-06-03 at 1 22 27 PM" src="https://user-images.githubusercontent.com/38776747/171934852-2c3c72b5-ab97-4bce-b697-9092285904d5.png">

I will work on this when I have some time.
